UI/newuser.html

```diff
@@ -181,9 +184,12 @@
                     func(username, successResponseValue);
                 },
                 error: function (errorValue) {
-                    if (errorValue.status === 400) {
+                    if (errorValue.status >= 400) {
                         alert('Tarkista syöttämäsi tiedot.');
+                    } else if (errorValue.status === 200) {
+                        func(username, successResponseValue);
```

Comment:
Can this be possible?
If the status code is 200 it will go to success function above and never enter this one, right?

Reply:
from what i understand success/error isn't only defined by the status. The response also needs to conform with the dataType of the request. This is not defined because we are sending form data, so setting it breaks the back end. Maybe there is some serialization that would allow us to set the data type, couldn't find it directly. If someone knows, please change it!

Reply:
From what I read it is possible to go to the error function with status code 200 IF ajax can't parse the response(for example, can be expecting a json and the server sended a XML). Since we are always expecting a json response it would be impossible, I think, to enter this part of the code.